src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -227,36 +226,11 @@ public void WriteIdentity(string key, long value)
 
             public void Dispose()
             {
-                try
-                {
-                    if (_identities.Count == 0)
-                        return;
-
-                    _database.TxMerger.Enqueue(new UpdateIdentitiesCommand(_identities, _database)).Wait();
-                }
-                finally
-                {
-                    _returnContext?.Dispose();
-                }
-            }
-
-            private class UpdateIdentitiesCommand : TransactionOperationsMerger.MergedTransactionCommand
-            {
-                private readonly Dictionary<string, long> _identities;
-                private readonly DocumentDatabase _database;
-
-                public UpdateIdentitiesCommand(Dictionary<string, long> identities, DocumentDatabase database)
-                {
-                    _identities = identities;
-                    _database = database;
-                }
-
-                public override int Execute(DocumentsOperationContext context)
-                {
-                    _database.DocumentsStorage.Identities.Update(context, _identities);
+                if (_identities.Count == 0)
+                    return;
 
-                    return 1;
-                }
+                //fire and forget, do not hold-up smuggler operations waiting for Raft command
+                _database.ServerStore.SendToLeaderAsync(new UpdateClusterIdentityCommand(_database.Name, _identities)).Wait();
```

Comment:
You _aren't_ doing fire and forget here.
And the smuggler is mostly async anyway. Move this behavior to FlushAsync()